### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by ef4**
This runs the project through prettier and adds a prettier check to github actions.

I have a more substantial PR coming behind this one, but inconsistent styling was making it hard to produce one that is easy to review.

### CODE REVIEW

**Comment by rwjblue** [comment]
Looks like it is failing the prettier check you added

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.d.ts
```
Comment (rwjblue) [inline .prettierignore:1]: We can bring this back, at least for our own d.ts files (seems fine to do in a follow up though)

File: .prettierrc.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+module.exports = {
+  singleQuote: true,
+  trailingComma: 'es5',
+  printWidth: 120,
+  arrowParens: 'avoid',
```
Comment (rwjblue) [inline .prettierrc.js:7]: I prefer the other style, but this probably more closely aligned with what is in the repo already

File: .prettierignore (hunk 1 above)
Comment (ef4) [reply .prettierignore:1]: The d.ts files inside here are entirely build artifacts. The reason this failed in CI but not locally is that I had no run tsc before linting.

Comment (rwjblue) [reply .prettierignore:1]: gotcha

**Comment by rwjblue** [comment]
We could also change CI a bit to only run linting on one platform.

**Review by rwjblue — APPROVED** [review]

**Pull request merged**